### ISSUE TITLE
Passing of time in games

### ISSUE THREAD

**Issue by adamantinum** [rafaelvcaetano/melonDS-android#176]
Device model: Poco F1

Android Version: LineageOS 18

Problem: There are issues connected with the passing of time. The emulator shows the correct system time and date, but in games where the passing of time is relevant there are issues (e.g. in pokemon HeartGold  apicorns should be generated everyday, instead it seems thery are generated after 24 hours of play, berries growth etc.)
Thank you in advance.

**Comment by butterfly2706** [comment]
I also had this issue the other day. What I did is test the 0.9.1 pc version which is the version the android port is based. As it turned out, having the Randomize MAC address option in the PC version messes up the time continuity of the game. Simple test I made is successfully migrate pokemon in platinum, close melonds, wait for a few minutes then open the application. When Randomize MAC Address is enabled, it says that the game time and the DS time are not in sync. If disabled, it does not.
The issue though is when Randomize MAC Address is disabled, two instances can't find each other via wireless local
Since the android port is based on the 0.9.1 version, I assume that the android port has this option enabled by default without the option of disabling.

**Comment by rafaelvcaetano** [comment]
@butterfly2706 thanks for the in-depth analysis. You are absolutely correct. Since local multiplayer was enabled, the Rabdomized MAC option is always enabled. I'll try to fix this for the next release.

**Comment by butterfly2706** [comment]
@rafaelvcaetano I'd love to see it be an option that can be toggled on/off since I love the local multiplayer capability when combining the app with an app cloner

**Comment by adamantinum** [comment]
Thank you all for the interest!

**Comment by rafaelvcaetano** [comment]
Try the latest [Beta 1.4.0](https://github.com/rafaelvcaetano/melonDS-android/releases/latest) release. This problem should now be fixed unless you enable the "Randomize MAC address" option.

**Comment by butterfly2706** [comment]
Just tested right now. Confirm that time passage now works if the option is toggled off

**Comment by adamantinum** [comment]
I should have done something wrong because it doesn't seem working...

**Comment by butterfly2706** [comment]
@adamantinum turn off randomize
![Screenshot_2021-03-29-21-01-02-16](https://user-images.githubusercontent.com/80578498/112840512-02d55780-90d2-11eb-9fe8-f8c2a55e3d98.jpg)

It should work

**Comment by adamantinum** [comment]
Yes, already done, I don't know why it isn't working. I tried to reinstall the app too.

**Comment by butterfly2706** [comment]
Do you use Custom BIOS? I do so I am not 100% this will work

But try this method

Plant an Oran Berry.
Save your game (ingame)
Close the app
Advance your phone time by 4 hours
Open the app and Load the game
if it newly planted gets a sprout, it means it is working

**Comment by adamantinum** [comment]
Not working... O.O

**Comment by butterfly2706** [comment]
That's weird... are you using a Custom BIOS?
If not, try to generate a new MAC address then do the steps above. See if it will work

**Comment by adamantinum** [comment]
Nothing to do, it's curious ahah

**Comment by corigne** [comment]
Hey! I was wondering, it's been a while. Is there any movement on this? I see suggestions for turning off randomization of the MAC address. Is there a possible downside to this or perhaps a better workaround?
